### PR TITLE
Update/k8s 1.30 deps

### LINKED ISSUE #4569: update kubernetes dependency to 1.30
Kubernetes 1.30 is released  https://kubernetes.io/blog/2024/04/17/kubernetes-v1-30-release/ let's make use of it in cephcsi

- [ ]  Make Kubernetes E2E test mandatory in CI Run
- [x] Update Mergify rules to consider tests on kube 1.30
- [x] Drop un maintained kube releases
https://github.com/ceph/ceph-csi/pull/4589
- [x] go dependency update
https://github.com/ceph/ceph-csi/pull/4593
- [ ] Readme update

### PR DESCRIPTION
**Pull request by iPraveenParihar**
# Describe what this PR does #

- cleanup: drop unmaintained 1.26 k8s release

- cleanup: add rules to run tests on k8s 1.30 release

## Related issues ##

Fixes: #4569 

**Checklist:**

* [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://github.com/ceph/ceph-csi/blob/devel/docs/development-guide.md#commit-messages).
* [x] Reviewed the developer guide on [Submitting a Pull Request](https://github.com/ceph/ceph-csi/blob/devel/docs/development-guide.md#development-workflow)
* [x] [Pending release notes](https://github.com/ceph/ceph-csi/blob/devel/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next major release.
* [x] Documentation has been updated, if necessary.
* [x] Unit tests have been added, if necessary.
* [x] Integration tests have been added, if necessary.

---

<details>
<summary>Show available bot commands</summary>

These commands are normally not required, but in case of issues, leave any of
the following bot commands in an otherwise empty comment in this PR:

* `/retest ci/centos/<job-name>`: retest the `<job-name>` after unrelated
  failure (please report the failure too!)

</details>

### CODE REVIEW

**Review by Madhu-1 — COMMENTED** [review]
Need to update https://github.com/ceph/ceph-csi/blob/devel/.github/workflows/pull-request-commentor.yaml as well

File: .github/workflows/pull-request-commentor.yaml [hunk 1]
```diff
@@ -19,12 +19,8 @@ jobs:
     strategy:
       matrix:
         branch: [release-v3.10, release-v3.11, devel]
-        k8s: ["1.26", "1.27", "1.28", "1.29"]
+        k8s: ["1.27", "1.28", "1.29"]
```
Comment (nixpanic) [inline .github/workflows/pull-request-commentor.yaml]: release-v3.10 is still maintained and needs to be run against 1.26.

File: README.md [hunk 2]
```diff
@@ -56,7 +56,7 @@ environments.
 
 | Ceph CSI Version | Container Orchestrator Name | Version Tested     |
 | -----------------| --------------------------- | -------------------|
-| v3.11.0          | Kubernetes                  | v1.26, v1.27, v1.28, v1.29|
+| v3.11.0          | Kubernetes                  | v1.27, v1.28, v1.29|
```
Comment (nixpanic) [inline README.md]: (only) this part needs to be backported to release-v3.11.

**Comment by nixpanic** [comment]
The pull-request-commented change is posted in #4589

**Comment by iPraveenParihar** [comment]
Closing as changes are included in https://github.com/ceph/ceph-csi/pull/4589

**Pull request closed without merging**